extensions/resource-deployment/src/ui/validation/validations.ts

```diff
@@ -0,0 +1,227 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { EOL } from 'os';
+import { throwUnless } from '../../common/utils';
+import * as loc from '../../localizedConstants';
+import { InputComponent } from '../modelViewUtils';
+import { RadioGroupLoadingComponentBuilder } from '../radioGroupLoadingComponentBuilder';
+
+export interface ValidationState {
+	valid: boolean;
+	message?: string;
+}
+
+export type Validator = () => Promise<ValidationState>;
+export type VariableValueGetter = (variable: string) => (Promise<string | number | undefined | null> | Promise<string | undefined | null>);
+export type ValueGetter = () => (Promise<string | number | undefined | null> | Promise<string | undefined | null>);
+
+export const enum ValidationType {
+	IsInteger = 'is_integer',
+	Regex = 'regex_match',
+	LessThanOrEquals = '<=',
```

Comment:
LessThanOrEqualTo and GreaterThanOrEuqalTo

Reply:
done